File: LIS.Web/src/app/models/invoice-header.ts
export interface InvoiceHeader {
  id: number;
  invoiceNumber: string;
  patientId?: number;
  mrn?: string;
  admissionNumber?: string;
  invoiceDate: Date;
  totalAmount: number;
  paidAmount?: number;
  balanceAmount?: number;
  status: string;
  isDeleted: boolean;
  createdDate?: Date;
  modifiedDate?: Date;
  createdBy?: string;
  modifiedBy?: string;
  notes?: string;
  dueDate?: Date;
  paidDate?: Date;
}

export interface CreateInvoiceHeaderRequest {
  patientId?: number;
  mrn?: string;
  admissionNumber?: string;
  invoiceDate: Date;
  notes?: string;
  dueDate?: Date;
}

export interface UpdateInvoiceHeaderRequest {
  id: number;
  totalAmount?: number;
  paidAmount?: number;
  balanceAmount?: number;
  status?: string;
  notes?: string;
  dueDate?: Date;
  paidDate?: Date;
}

File: LIS.Api/Models/InvoiceHeader.cs
namespace LIS.Api.Models
{
    public class InvoiceHeader
    {
        public int Id { get; set; }
        public string InvoiceNumber { get; set; } = string.Empty;
        public int? PatientId { get; set; }
        public string? MRN { get; set; }
        public string? AdmissionNumber { get; set; }
        public DateTime InvoiceDate { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal? PaidAmount { get; set; }
        public decimal? BalanceAmount { get; set; }
        public string Status { get; set; } = string.Empty;
        public bool IsDeleted { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public string? CreatedBy { get; set; }
        public string? ModifiedBy { get; set; }
        public string? Notes { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? PaidDate { get; set; }
    }

    public class CreateInvoiceHeaderRequest
    {
        public int? PatientId { get; set; }
        public string? MRN { get; set; }
        public string? AdmissionNumber { get; set; }
        public DateTime InvoiceDate { get; set; }
        public string? Notes { get; set; }
        public DateTime? DueDate { get; set; }
    }

    public class UpdateInvoiceHeaderRequest
    {
        public int Id { get; set; }
        public decimal? TotalAmount { get; set; }
        public decimal? PaidAmount { get; set; }
        public decimal? BalanceAmount { get; set; }
        public string? Status { get; set; }
        public string? Notes { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? PaidDate { get; set; }
    }
}














